### ISSUE TITLE
G502 SE Hero Gaming Mouse Not Showing Up

### ISSUE THREAD

**Issue by pfred1** [pwr-Solaar/Solaar#2111]
I am a new Solaar user so it is not working for me. I worked through the documentation and satisfied all of the dependencies. The script does run. When it does the selection box is empty. I have a Logitech mouse that looks supported by "G502 SE Hero Gaming Mouse C08B.txt" the device ID matches lsusb. I am trying to use the github version 1.1.9-28-g12f9c01 that I cloned about an hour ago. So a fairly recent version. 

Any hints that could get this working more like I expected it to would be appreciated. 

Here's what I have done so far:
I satisfied all of the dependencies as far as Python modules and distribution packages go.
I loaded the hid-logitech-hidpp and hid-logitech-dj kernel modules.
I put that 42-logitech-unify-permissions.rules udev file in /etc/udev/rules.d and loaded it with #  udevadm control --reload-rules.

Then I ran the program and it came up blank. There's no errors or anything. So I am at a loss as to what else I can do.

What I expected:
I thought I'd see my mouse show up in the device selection box.

**Comment by pfps** [comment]
Run `solaar -ddd show` and post the output.  Also provide the output of `lsusb`.  The mouse should be connected and active.

**Comment by pfred1** [comment]
There's something to work with. Apparently I cannot access the device.
I am going to have to look into how to fix that. If you know I am open to suggestions. I off to the Googler now though.


$  bin/solaar -ddd
2023-08-03 16:29:08,746,746     INFO [MainThread] root: version 1.1.9-28-g12f9c01, language en_US (UTF-8)
2023-08-03 16:29:08,869,869     INFO [MainThread] logitech_receiver.diversion: GDK Keymap set up
2023-08-03 16:29:08,881,881    DEBUG [MainThread] solaar.ui.tray: using StatusIcon
2023-08-03 16:29:08,883,883     INFO [MainThread] solaar.upower: connected to system dbus, watching for suspend/resume events
2023-08-03 16:29:08,905,905    DEBUG [MainThread] solaar.ui: startup registered=True, remote=False
2023-08-03 16:29:08,906,906    DEBUG [AsyncUI] solaar.tasks: started
2023-08-03 16:29:08,906,906     INFO [MainThread] solaar.ui.notify: starting desktop notifications
2023-08-03 16:29:08,941,941    DEBUG [MainThread] solaar.ui.icons: sys.path[0] = /home/pfred1/build/Solaar/Solaar/lib
2023-08-03 16:29:08,941,941    DEBUG [MainThread] solaar.ui.icons: looking for icons in /home/pfred1/build/Solaar/Solaar/icons
2023-08-03 16:29:08,941,941    DEBUG [MainThread] solaar.ui.icons: looking for icons in /home/pfred1/build/Solaar/Solaar/share/solaar/icons
2023-08-03 16:29:08,941,941    DEBUG [MainThread] solaar.ui.icons: icon theme paths: ['/home/pfred1/build/Solaar/Solaar/share/solaar/icons', '/home/pfred1/.local/share/icons', '/home/pfred1/.icons', '/usr/local/share/icons', '/usr/share/icons', '/usr/local/share/pixmaps', '/usr/share/pixmaps']
2023-08-03 16:29:08,989,989     INFO [MainThread] solaar.listener: starting receiver listening threads
2023-08-03 16:29:09,135,135     INFO [MainThread] hidapi.udev: Found device BID 0003 VID 0000046D PID 0000C08B HID++ True True USB 1 1
2023-08-03 16:29:09,135,135     INFO [MainThread] solaar.listener: receiver event add DeviceInfo(path='/dev/hidraw1', bus_id=3, vendor_id='046D', product_id='C08B', interface=1, driver='hid-generic', manufacturer=None, product=None, serial='1273317A3133', release=None, isDevice=True, hidpp_short=True, hidpp_long=True)
2023-08-03 16:29:09,135,135     INFO [MainThread] hidapi.udev: OPEN PATH /dev/hidraw1
2023-08-03 16:29:09,135,135     INFO [MainThread] hidapi.udev: OPEN PATH FAILED /dev/hidraw1 ERROR 13 [Errno 13] Permission denied: '/dev/hidraw1'
2023-08-03 16:29:09,235,235     INFO [MainThread] hidapi.udev: OPEN PATH FAILED /dev/hidraw1 ERROR 13 [Errno 13] Permission denied: '/dev/hidraw1'
2023-08-03 16:29:09,336,336     INFO [MainThread] hidapi.udev: OPEN PATH FAILED /dev/hidraw1 ERROR 13 [Errno 13] Permission denied: '/dev/hidraw1'
2023-08-03 16:29:09,436,436  WARNING [MainThread] solaar.listener: failed to open DeviceInfo(path='/dev/hidraw1', bus_id=3, vendor_id='046D', product_id='C08B', interface=1, driver='hid-generic', manufacturer=None, product=None, serial='1273317A3133', release=None, isDevice=True, hidpp_short=True, hidpp_long=True)
2023-08-03 16:29:09,437,437    DEBUG [MainThread] solaar.ui: activate
2023-08-03 16:29:15,348,348    DEBUG [MainThread] solaar.ui: shutdown
2023-08-03 16:29:15,349,349    DEBUG [AsyncUI] solaar.tasks: stopped
2023-08-03 16:29:15,350,350     INFO [MainThread] solaar.ui.notify: stopping desktop notifications


$  lsusb
Bus 001 Device 002: ID 046d:c08b Logitech, Inc. G502 SE HERO Gaming Mouse

**Comment by pfps** [comment]
It looks as if your Solaar udev rule is not firing.  Check for appropriate ownership (root) and permissions (no write permissions for anyone except root).  

Run `getfacl /dev/hidraw*`.  If the rule is working correctly you should see a line like:
```
user:<your user name>:rw-
```
for the node for your mouse.

**Comment by pfred1** [comment]
I did some poking around on the net and the easy solution was rebooting. Now my mouse is in Solaar. Though some suggested just unplugging the mouse and plugging it back in may have worked too. But it's plugged into the back of the PC so I didn't want to do that. Why # udevadm control --reload-rules didn't work is beyond my pay scale. I did find there's two 42-logitech-unify-permissions.rules files in the source tree. I used the one the readme said to use. But the other one has another line in it.
$  diff ./rules.d-uinput/42-logitech-unify-permissions.rules ./rules.d/42-logitech-unify-permissions.rules
6d5
< KERNEL=="uinput", SUBSYSTEM=="misc", TAG+="uaccess", OPTIONS+="static_node=uinput"

I don't know. The one I'm using works now? Which is better was something I was hazy on with everything. So I opted for the one that the readme said to use. But maybe the longer one with the extra line would have worked? I've no idea what any of that means.
